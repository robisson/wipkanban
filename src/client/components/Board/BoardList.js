// @flow
import * as React from "react";
import Grid from "@material-ui/core/Grid";
import Typography from "@material-ui/core/Typography";
import DeveloperBoardIcon from "@material-ui/icons/DeveloperBoard";
import MiniBoard from "./MiniBoard";
import Tooltip from "@material-ui/core/Tooltip";
import AddIcon from "@material-ui/icons/AddCircle";
import IconButton from "@material-ui/core/IconButton";

type Props = {
  boards: Array<Object>
};

const BoardList = ({ boards }: Props) => {
  let boardsRendered = boards.map(board => (
    <MiniBoard key={board._id} {...board} />
  ));

  return (
    <div>
      <div>
        <Typography color="secondary" variant="h6" gutterBottom>
          <DeveloperBoardIcon /> My boards
          <Tooltip title="Add team">
            <IconButton color="default" aria-label="Add board">
              <AddIcon />
            </IconButton>
          </Tooltip>
        </Typography>
      </div>
      <Grid container justify="flex-start" wrap="wrap" spacing={0}>
        <Grid item lg={6}>
          {boardsRendered}
        </Grid>
      </Grid>
    </div>
  );
};

export default BoardList;
