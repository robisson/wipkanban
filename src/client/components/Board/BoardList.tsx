// @flow
import * as React from "react";
import Grid from "@material-ui/core/Grid";
import Typography from "@material-ui/core/Typography";
import DeveloperBoardIcon from "@material-ui/icons/DeveloperBoard";
import MiniBoard from "./MiniBoard";
import Tooltip from "@material-ui/core/Tooltip";
import AddBoard from "./AddBoardContainer";

interface IBoardListProps {
  boards: Array<Object>;
}

const BoardList: React.StatelessComponent<any> = ({
  boards
}: IBoardListProps) => {
  let boardsRendered = boards.map((board: any) => (
    <Grid key={board._id} item lg={6}>
      <MiniBoard key={board._id} {...board} />
    </Grid>
  ));

  return (
    <div>
      <div>
        <Typography color="secondary" variant="h6" gutterBottom>
          <DeveloperBoardIcon /> My boards
          <Tooltip title="Add team">
            <AddBoard />
          </Tooltip>
        </Typography>
      </div>
      <Grid container justify="flex-start" wrap="wrap" spacing={8}>
        {boardsRendered}
      </Grid>
    </div>
  );
};

export default BoardList;
