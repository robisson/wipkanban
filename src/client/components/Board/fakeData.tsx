let columns = [
  {
    id: 4,
    title: "A fazer",
    hColumns: [],
    vColumns: [],
    cards: [
      {
        id: 31,
        title:
          "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card d" +
          "o quadro kanban"
      }
    ]
  },
  {
    id: 1,
    title: "Executando",
    hColumns: [
      {
        id: 2,
        title: "Chamados",
        hColumns: [],
        vColumns: [
          {
            id: 4,
            title: "Parado",
            hColumns: [],
            vColumns: [],
            cards: []
          },
          {
            id: 5,
            title: "Executando",
            hColumns: [],
            vColumns: [],
            cards: [
              {
                id: 35,
                title:
                  "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card d" +
                  "o quadro kanban"
              }
            ]
          }
        ],
        cards: []
      },
      {
        id: 3,
        title: "Projetos",
        hColumns: [],
        vColumns: [
          {
            id: 41,
            title: "Parado",
            hColumns: [],
            vColumns: [],
            cards: [
              {
                id: 36,
                title:
                  "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card d" +
                  "o quadro kanban"
              }
            ]
          },
          {
            id: 51,
            title: "Executando",
            hColumns: [],
            vColumns: [],
            cards: []
          }
        ],
        cards: []
      }
    ],
    vColumns: [],
    cards: []
  },
  {
    id: 9,
    title: "Em teste",
    hColumns: [],
    vColumns: [],
    cards: [
      {
        id: 32,
        title:
          "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card d" +
          "o quadro kanban"
      }
    ]
  },
  {
    id: 10,
    title: "Concluído",
    hColumns: [],
    vColumns: [],
    cards: [
      {
        id: 33,
        title:
          "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card d" +
          "o quadro kanban"
      },
      {
        id: 34,
        title:
          "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card d" +
          "o quadro kanban"
      }
    ]
  }
];

export default columns;
