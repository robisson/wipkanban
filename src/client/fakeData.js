export const boards = [
  {
    _id: 12121212,
    name: "Minhas atividades",
    createdBy: "Robisson Oliveira",
    createdAt: "today 12:00pm",
    leadTime: "10",
    cycleTime: "10",
    throuput: "10",
    columns: [
      {
        id: 4,
        title: "A fazer",
        hColumns: [],
        vColumns: [],
        cards: [
          {
            id: 31,
            title:
              "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card d" +
              "o quadro kanban"
          }
        ]
      },
      {
        id: 1,
        title: "Executando",
        hColumns: [
          {
            id: 2,
            title: "Chamados",
            hColumns: [],
            vColumns: [
              {
                id: 4,
                title: "Parado",
                hColumns: [],
                vColumns: [],
                cards: []
              },
              {
                id: 5,
                title: "Executando",
                hColumns: [],
                vColumns: [],
                cards: [
                  {
                    id: 35,
                    title:
                      "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card d" +
                      "o quadro kanban"
                  }
                ]
              }
            ],
            cards: []
          },
          {
            id: 3,
            title: "Projetos",
            hColumns: [],
            vColumns: [
              {
                id: 41,
                title: "Parado",
                hColumns: [],
                vColumns: [],
                cards: [
                  {
                    id: 36,
                    title:
                      "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card d" +
                      "o quadro kanban"
                  }
                ]
              },
              {
                id: 51,
                title: "Executando",
                hColumns: [],
                vColumns: [],
                cards: []
              }
            ],
            cards: []
          }
        ],
        vColumns: [],
        cards: []
      },
      {
        id: 9,
        title: "Em teste",
        hColumns: [],
        vColumns: [],
        cards: [
          {
            id: 32,
            title:
              "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card d" +
              "o quadro kanban"
          }
        ]
      },
      {
        id: 10,
        title: "Concluído",
        hColumns: [],
        vColumns: [],
        cards: [
          {
            id: 33,
            title:
              "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card d" +
              "o quadro kanban"
          },
          {
            id: 34,
            title:
              "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card d" +
              "o quadro kanban"
          }
        ]
      }
    ]
  }
];
