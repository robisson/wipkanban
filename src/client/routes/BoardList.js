// @flow
import * as React from "react";
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import DeveloperBoardIcon from '@material-ui/icons/DeveloperBoard';
import MiniBoard from './MiniBoard';

const BoardList = () => {
  return (
    <div>
      <div>
        <Typography color="secondary" variant="title" gutterBottom>
          <DeveloperBoardIcon/>{' '}
          My boards
        </Typography>
      </div>
      <Grid container justify="flex-start" wrap="nowrap" spacing={0}>
        <Grid item lg={6}>
          <MiniBoard/>
        </Grid>
      </Grid>
    </div>
  )
};

export default BoardList;
