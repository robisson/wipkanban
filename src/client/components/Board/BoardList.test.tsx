import React from "react";
import BoardList from "./BoardList";
import { createShallow } from "@material-ui/core/test-utils";
import Grid from "@material-ui/core/Grid";
import Typography from "@material-ui/core/Typography";
import { boards } from "../../fakeData";

describe("BoardList Component", () => {
  let shallow: any;

  beforeEach(() => {
    shallow = createShallow();
  });

  it("should render a div", () => {
    let wrapper = shallow(<BoardList boards={boards} />);
    expect(wrapper.type()).toEqual("div");
    expect(wrapper).toMatchSnapshot();
  });

  it("should render 1 Typogragh and 2 Grid", () => {
    let wrapper = shallow(<BoardList boards={boards} />);
    expect(wrapper.find(Typography).length).toEqual(1);
    expect(wrapper.find(Grid).length).toEqual(2);
    expect(wrapper).toMatchSnapshot();
  });
});
