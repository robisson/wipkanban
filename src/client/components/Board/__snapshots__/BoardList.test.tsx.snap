// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardList Component should render 1 Typogragh and 2 Grid 1`] = `
<div>
  <div>
    <WithStyles(ForwardRef(Typography))
      color="secondary"
      gutterBottom={true}
      variant="h6"
    >
      <DeveloperBoardIcon />
       My boards
      <WithStyles(ForwardRef(Tooltip))
        title="Add team"
      >
        <Connect(FormDialog) />
      </WithStyles(ForwardRef(Tooltip))>
    </WithStyles(ForwardRef(Typography))>
  </div>
  <WithStyles(ForwardRef(Grid))
    container={true}
    justify="flex-start"
    spacing={8}
    wrap="wrap"
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      key="12121212"
      lg={6}
    >
      <WithStyles(MediaControlCard)
        _id={12121212}
        columns={
          Array [
            Object {
              "cards": Array [
                Object {
                  "id": 31,
                  "title": "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card do quadro kanban",
                },
              ],
              "hColumns": Array [],
              "id": 4,
              "title": "A fazer",
              "vColumns": Array [],
            },
            Object {
              "cards": Array [],
              "hColumns": Array [
                Object {
                  "cards": Array [],
                  "hColumns": Array [],
                  "id": 2,
                  "title": "Chamados",
                  "vColumns": Array [
                    Object {
                      "cards": Array [],
                      "hColumns": Array [],
                      "id": 4,
                      "title": "Parado",
                      "vColumns": Array [],
                    },
                    Object {
                      "cards": Array [
                        Object {
                          "id": 35,
                          "title": "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card do quadro kanban",
                        },
                      ],
                      "hColumns": Array [],
                      "id": 5,
                      "title": "Executando",
                      "vColumns": Array [],
                    },
                  ],
                },
                Object {
                  "cards": Array [],
                  "hColumns": Array [],
                  "id": 3,
                  "title": "Projetos",
                  "vColumns": Array [
                    Object {
                      "cards": Array [
                        Object {
                          "id": 36,
                          "title": "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card do quadro kanban",
                        },
                      ],
                      "hColumns": Array [],
                      "id": 41,
                      "title": "Parado",
                      "vColumns": Array [],
                    },
                    Object {
                      "cards": Array [],
                      "hColumns": Array [],
                      "id": 51,
                      "title": "Executando",
                      "vColumns": Array [],
                    },
                  ],
                },
              ],
              "id": 1,
              "title": "Executando",
              "vColumns": Array [],
            },
            Object {
              "cards": Array [
                Object {
                  "id": 32,
                  "title": "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card do quadro kanban",
                },
              ],
              "hColumns": Array [],
              "id": 9,
              "title": "Em teste",
              "vColumns": Array [],
            },
            Object {
              "cards": Array [
                Object {
                  "id": 33,
                  "title": "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card do quadro kanban",
                },
                Object {
                  "id": 34,
                  "title": "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card do quadro kanban",
                },
              ],
              "hColumns": Array [],
              "id": 10,
              "title": "Concluído",
              "vColumns": Array [],
            },
          ]
        }
        createdAt="today 12:00pm"
        createdBy="Robisson Oliveira"
        cycleTime="10"
        key="12121212"
        leadTime="10"
        name="Minhas atividades"
        throuput="10"
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</div>
`;

exports[`BoardList Component should render a div 1`] = `
<div>
  <div>
    <WithStyles(ForwardRef(Typography))
      color="secondary"
      gutterBottom={true}
      variant="h6"
    >
      <DeveloperBoardIcon />
       My boards
      <WithStyles(ForwardRef(Tooltip))
        title="Add team"
      >
        <Connect(FormDialog) />
      </WithStyles(ForwardRef(Tooltip))>
    </WithStyles(ForwardRef(Typography))>
  </div>
  <WithStyles(ForwardRef(Grid))
    container={true}
    justify="flex-start"
    spacing={8}
    wrap="wrap"
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      key="12121212"
      lg={6}
    >
      <WithStyles(MediaControlCard)
        _id={12121212}
        columns={
          Array [
            Object {
              "cards": Array [
                Object {
                  "id": 31,
                  "title": "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card do quadro kanban",
                },
              ],
              "hColumns": Array [],
              "id": 4,
              "title": "A fazer",
              "vColumns": Array [],
            },
            Object {
              "cards": Array [],
              "hColumns": Array [
                Object {
                  "cards": Array [],
                  "hColumns": Array [],
                  "id": 2,
                  "title": "Chamados",
                  "vColumns": Array [
                    Object {
                      "cards": Array [],
                      "hColumns": Array [],
                      "id": 4,
                      "title": "Parado",
                      "vColumns": Array [],
                    },
                    Object {
                      "cards": Array [
                        Object {
                          "id": 35,
                          "title": "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card do quadro kanban",
                        },
                      ],
                      "hColumns": Array [],
                      "id": 5,
                      "title": "Executando",
                      "vColumns": Array [],
                    },
                  ],
                },
                Object {
                  "cards": Array [],
                  "hColumns": Array [],
                  "id": 3,
                  "title": "Projetos",
                  "vColumns": Array [
                    Object {
                      "cards": Array [
                        Object {
                          "id": 36,
                          "title": "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card do quadro kanban",
                        },
                      ],
                      "hColumns": Array [],
                      "id": 41,
                      "title": "Parado",
                      "vColumns": Array [],
                    },
                    Object {
                      "cards": Array [],
                      "hColumns": Array [],
                      "id": 51,
                      "title": "Executando",
                      "vColumns": Array [],
                    },
                  ],
                },
              ],
              "id": 1,
              "title": "Executando",
              "vColumns": Array [],
            },
            Object {
              "cards": Array [
                Object {
                  "id": 32,
                  "title": "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card do quadro kanban",
                },
              ],
              "hColumns": Array [],
              "id": 9,
              "title": "Em teste",
              "vColumns": Array [],
            },
            Object {
              "cards": Array [
                Object {
                  "id": 33,
                  "title": "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card do quadro kanban",
                },
                Object {
                  "id": 34,
                  "title": "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card do quadro kanban",
                },
              ],
              "hColumns": Array [],
              "id": 10,
              "title": "Concluído",
              "vColumns": Array [],
            },
          ]
        }
        createdAt="today 12:00pm"
        createdBy="Robisson Oliveira"
        cycleTime="10"
        key="12121212"
        leadTime="10"
        name="Minhas atividades"
        throuput="10"
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</div>
`;
